src/mesh/interpolation/xz/impls/petsc_hermite_spline_xz.cxx

```diff
@@ -322,31 +321,30 @@ void XZHermiteSpline::calcWeights(const Field3D& delta_x, const Field3D& delta_z
  *   (i, j+1, k+2)	h11_z / 2
  */
 std::vector<ParallelTransform::PositionsAndWeights>
-XZHermiteSpline::getWeightsForYApproximation(int i, int j, int k, int yoffset) {
+XZPetscHermiteSpline::getWeightsForYApproximation(int i, int j, int k, int yoffset) {
   const int nz = localmesh->LocalNz;
   const int k_mod = k_corner(i, j, k);
   const int k_mod_m1 = (k_mod > 0) ? (k_mod - 1) : (nz - 1);
   const int k_mod_p1 = (k_mod == nz) ? 0 : k_mod + 1;
   const int k_mod_p2 = (k_mod_p1 == nz) ? 0 : k_mod_p1 + 1;
 
   return {{i, j + yoffset, k_mod_m1, -0.5 * h10_z(i, j, k)},
-          {i, j + yoffset, k_mod, h00_z(i, j, k) - 0.5 * h11_z(i, j, k)},
-          {i, j + yoffset, k_mod_p1, h01_z(i, j, k) + 0.5 * h10_z(i, j, k)},
+          {i, j + yoffset, k_mod, h00_z(i, j, k) - (0.5 * h11_z(i, j, k))},
+          {i, j + yoffset, k_mod_p1, h01_z(i, j, k) + (0.5 * h10_z(i, j, k))},
           {i, j + yoffset, k_mod_p2, 0.5 * h11_z(i, j, k)}};
 }
 
-Field3D XZHermiteSpline::interpolate(const Field3D& f, const std::string& region) const {
+Field3D XZPetscHermiteSpline::interpolate(const Field3D& f,
+                                          const std::string& region) const {
```

Comment:
@dschwoerer `region` isn't actually used here and I don't really understand why not. This technically makes this version behave differently to the other implementations. Should they actually all use `RGN_NOY/YPAR_<N>`?

It doesn't look like we ever actually call `interpolate` with a region anywhere, so perhaps it would be better to remove the argument and fix all the implementations?